### PR TITLE
Limit public visibility of virtio_gen crate by restricting bindgen arguments

### PR DESCRIPTION
**Pull request by avr1**
# Reason for This PR
 In an effort to contribute to issue #607, this pull request removes most of the wildcard usages of the crate, and then updates the bindgen commands used to generate these files to return only the types that are used in other parts of the codebase.

As an aside, this is my first contribution, so please excuse any mistakes in the process. Looking forward to contributing more!

## Description of Changes

Modifications to files that use elements in virtio_gen, where the actual types and variables used from that file are explicitly mentioned in the header. Furthermore, the 3 core files in virtio_gen are modified, each one just storing the values that correspond to the usages.

- [ ] This functionality can be added in [`rust-vmm`][1].

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The issue which led to this PR has a clear conclusion.
- [x] This PR follows the solution outlined in the related issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes follow the [Runbook for Firecracker API changes][2].
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

[1]: https://github.com/rust-vmm
[2]: ../docs/api-change-runbook.md

### CODE REVIEW

File: src/devices/src/virtio/block/device.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ use std::sync::Arc;
 use logger::{error, warn, IncMetric, METRICS};
 use rate_limiter::{BucketUpdate, RateLimiter};
 use utils::eventfd::EventFd;
-use virtio_gen::virtio_blk::*;
+use virtio_gen::virtio_blk::{VIRTIO_BLK_ID_BYTES, VIRTIO_F_VERSION_1, VIRTIO_BLK_F_FLUSH, VIRTIO_BLK_F_RO, };
```
Comment (alindima) [inline src/devices/src/virtio/block/device.rs]: You should squash this commit into the previous one. Ideally we should have two commits, one removing the wildcard usage and another one updating the bindings

File: src/devices/src/virtio/block/device.rs [hunk 2]
```diff
@@ -18,7 +18,7 @@ use std::sync::Arc;
 use logger::{error, warn, IncMetric, METRICS};
 use rate_limiter::{BucketUpdate, RateLimiter};
 use utils::eventfd::EventFd;
-use virtio_gen::virtio_blk::{VIRTIO_BLK_ID_BYTES, VIRTIO_F_VERSION_1, VIRTIO_BLK_F_FLUSH, VIRTIO_BLK_F_RO, };
+use virtio_gen::virtio_blk::{VIRTIO_BLK_ID_BYTES, VIRTIO_F_VERSION_1, VIRTIO_BLK_F_FLUSH, VIRTIO_BLK_F_RO};
```
Comment (alindima) [inline src/devices/src/virtio/block/device.rs:21]: please make sure every commit passes the style tests: `tools/devtool test -- integration_tests/style`

**Review by alindima — COMMENTED** [review]
Thanks for the contribution. Please make the appropriate changes to keep the unit tests passing. They are currently failing due to some unresolved imports caused by the removal of the wildcard imports.

You may run the unit tests by:
 `tools/devtool test -- integration_tests/build/test_unittests.py`

**Comment by dianpopa** [comment]
@avr1 PR looks good. Apart from Alin's suggestions, you also need to run the style tests: `tools/devtool test -- integration_tests/style/`

After you solve the failling tests, the PR should be good to go.

**Comment by alindima** [comment]
@avr1 need any help to push this over the finish line?

**Comment by avr1** [comment]
@alindima Thanks for reaching out! I just sent a message on Slack about the issue that I'm running into, can we continue the conversation there?

**Comment by alindima** [comment]
@avr1 need any help?
If you will, I can force-push over your branch so that we can have this merge-ready.

**Review by alindima — APPROVED** [review]

**Review by dianpopa — APPROVED** [review]

**Review by luminitavoicu — APPROVED** [review]

**Pull request merged**